Interactiviy API Docs: Merge Interactivity docs to keep consistency with other packages. (#59270)
WordPress/gutenberg: packages/interactivity/README.md

# Interactivity API

> **Note**
> This package enables the API shared at [Proposal: The Interactivity API – A better developer experience in building interactive blocks](https://make.wordpress.org/core/2023/03/30/proposal-the-interactivity-api-a-better-developer-experience-in-building-interactive-blocks/). As part of an [Open Source project](https://developer.wordpress.org/block-editor/getting-started/faq/#the-gutenberg-project), participation is encouraged in testing this API providing feedback at the [discussions in GitHub](https://github.com/WordPress/gutenberg/discussions/categories/interactivity-api).

The Interactivity API is available at WordPress Core from version 6.5 [Merge announcement](https://make.wordpress.org/core/2024/02/19/merge-announcement-interactivity-api/)

These Core blocks are already powered by thi API:

- Search
- Query
- Navigation
- File

## Installation

Install the module:

```bash
npm install @wordpress/interactivity --save

> ```

Reviewer: I know this was there in the previous docs but I would add a note here that this step is only required if you are building outside of WordPress... 

Within WordPress you don't need to install the package via NPM because it is bundled with WordPress core and all you need to do to ensure it is loaded is adding `wp-interactivity` to the dependency array of the module script. 

Which happens automatically when you use the dependency ectraction webpack plugin that is used in tools like wp-scripts.

- [Quick Start Guide](#quick-start-guide)
    - [1. Scaffold an interactive block](#1-scaffold-an-interactive-block)
    - [2. Generate the build](#2-generate-the-build)
    - [3. Use it in your WordPress installation ](#3-use-it-in-your-wordpress-installation)
- [Requirements of the Interactivity API](#requirements-of-the-interactivity-aPI)
    - [A local WordPress installation](#a-local-wordpress-installation)
    - [Latest vesion of Gutenberg](#latest-vesion-of-gutenberg)
    - [Node.js](#nodejs)
    - [Code requirements](#code-requirements)
        - [Add `interactivity` support to `block.json`](#add-interactivity-support-to-blockjson)
        - [Add `wp-interactive` directive to a DOM element](#add-wp-interactive-directive-to-a-dom-element)

#### 1. Scaffold an interactive block

A WordPress plugin that registers an interactive block (using the Interactivity API) by using a [template](https://www.npmjs.com/package/@wordpress/create-block-interactive-template) can be scaffolded with the `@wordpress/create-block` command.

```
npx @wordpress/create-block@latest my-first-interactive-block --template @wordpress/create-block-interactive-template
```

#### 2. Generate the build

> When the plugin folder is generated, the build process needs to be launched to get the final version of the interactive block that can be used from WordPress.

Reviewer: ```suggestion
When the plugin folder is generated, the build process needs to be launched to get a working version of the interactive block that can be used in WordPress.
```
